classes/export_webservices.php

```diff
@@ -0,0 +1,387 @@
+        }
+
+        return $webservices;
+
+    }
+
+
+    /**
+     * Retrieves indexed webservice records from the database.
+     *
+     * @return array An array of the indexed webservices.
+     */
+    private function get_indexed_webservice_records(): array {
+        global $DB;
+
+        // Get_records returns an object array where key for each object is the name of the webservice.
+        // Use array_values to change key to the index of each object so that we can filter based on $selectedWebserviceIndices.
+        $webservicesrecords = array_values($DB->get_records('external_functions', [], ''));
+
+        return $webservicesrecords;
+
+    }
+
+
+    /**
+     * Generates HTML for REST parameter description.
+     *
+     * @param  object $paramdescription Description object for the parameter.
+     * @param  string $paramstring      Parameter string to be formatted.
+     * @return mixed HTML formatted parameter description.
+     */
+    public function rest_param_description_html(object $paramdescription, string $paramstring): mixed {
+        $brakeline = <<<EOF
+
+
+        EOF;
+
+        // Description object is a list.
+        if ($paramdescription instanceof external_multiple_structure) {
+            $paramstring = $paramstring . '[0]';
+            $return      = $this->rest_param_description_html($paramdescription->content, $paramstring);
+            return $return;
+        } else if ($paramdescription instanceof external_single_structure) {
+            // Description object is an object.
+            $singlestructuredesc = '';
+            $initialparamstring  = $paramstring;
+            foreach ($paramdescription->keys as $attributname => $attribut) {
+                $paramstring          = $initialparamstring . '[' . $attributname . ']';
+                $singlestructuredesc .= $this->rest_param_description_html(
+                    $paramdescription->keys[$attributname],
+                    $paramstring
+                );
+            }
+
+            return $singlestructuredesc;
+        } else {
+            // Description object is a primary type (string, integer).
+            $paramstring = $paramstring . '=';
+            $type        = '';
+
+            switch ($paramdescription->type) {
+                    // 0 or 1 only for now
+                case PARAM_INT:
+                    $type = '{{INT}}';
+                    break;
+
+                case PARAM_BOOL:
+                    $type = '{{BOOL}}';
+                    break;
+
+                case PARAM_TEXT:
+                    $type = '{{TEXT}}';
+                    break;
+
+                case PARAM_ALPHA:
+                    $type = '{{ALPHA}}';
+                    break;
+
+                case PARAM_FLOAT;
+                    $type = '{{DOUBLE}}';
+                    break;
+
+                default:
+                    $type = '{{STRING}}';
+            }
+
+            return $paramstring.$type.$brakeline;
```

Comment:
These should be seperated with a space before and after the dot operator (treat it like a plus)